### ISSUE TITLE
Allow specifying a file sync limit

### ISSUE THREAD

**Issue by kiambogo** [gilt/s3-sftp-bridge#20]
In the case that an SFTP server has many files and a slower connection, we have seen issues where operating on all the files returned from listing the directory can cause `maximum call stack size exceeded` errors and cause the Lambda to timeout. From my experience, this actually limits the number of files that can be processed before the exception is thrown.

A resolution to this would be to allow for setting a buffer limit on the number of items from the directory list to process. 

I will try to get a PR for this soon.

**Comment by kiambogo** [comment]
@ashaseetharam

**Comment by ajshukury** [comment]
Yeah. I see this issue. I could submit a PR if this is where you wanna go? 
Also, would you care about bringing it under serverless framework instead of a cloud-formation template?
